pkg/cluster/deploystorage.go

```diff
@@ -72,9 +72,15 @@ func (m *manager) ensureResourceGroup(ctx context.Context) error {
 		ManagedBy: to.StringPtr(m.doc.OpenShiftCluster.ID),
 	}
 	if m.env.IsLocalDevelopmentMode() {
-		// TODO: ideally this should be removed, but removing it causes the
-		// purge script to ignore RGs in CI E2E.  Fix that at the same time.
-		group.ManagedBy = nil
+		rg, err := m.resourceGroups.Get(ctx, resourceGroup)
+		if err == nil {
+			group.Tags = rg.Tags
+		}
+		if group.Tags == nil {
+			group.Tags = map[string]*string{"purge": to.StringPtr("true")}
+		} else {
+			group.Tags["purge"] = to.StringPtr("true")
+		}
```

Comment:
Maybe something like this (didn't test it)?

```suggestion
		if group.Tags == nil {
			group.Tags = map[string]*string{}
		}
		group.Tags["purge"] = to.StringPtr("true")
```